src/test/java/com/recurly/v3/PagerTest.java

```diff
@@ -0,0 +1,150 @@
+package com.recurly.v3;
+
+import com.recurly.v3.fixtures.MockClient;
+import com.recurly.v3.fixtures.MyResource;
+import okhttp3.*;
+import okhttp3.Request;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+public class PagerTest {
+  @Test
+  public void testForEach() {
+    OkHttpClient mockOkHttpClient = getMockOkHttpClientSingleRequest(getResourceListJson());
+
+    final MockClient client = new MockClient("siteId", "apiKey", mockOkHttpClient);
+    Pager<MyResource> pager = client.listResources(null);
+    pager.forEach(resource -> assertNotNull(resource.getMyString()));
+  }
+
+  @Test
+  public void testEachItem() {
+    OkHttpClient mockOkHttpClient = getMockOkHttpClientSingleRequest(getResourceListJson());
+
+    final MockClient client = new MockClient("siteId", "apiKey", mockOkHttpClient);
+    Pager<MyResource> pager = client.listResources(null);
+    pager.eachItem(resource -> assertNotNull(resource.getMyString()));
+  }
+
+  @Test
+  public void testEmptyList() {
+    OkHttpClient mockOkHttpClient = getMockOkHttpClientSingleRequest(getEmptyListJson());
+
+    final MockClient client = new MockClient("siteId", "apiKey", mockOkHttpClient);
+    Pager<MyResource> pager = client.listResources(null);
+    assertEquals(0, pager.getData().size());
+
+    for (MyResource myResource : pager) {
+      myResource.getMyString(); // This should not throw NullPointerException
+    }
+    pager.forEach(myResource -> myResource.getMyString()); // This should not throw NullPointerException either
+  }
+
+  @Test
+  public void testMultiplePages() {
+    OkHttpClient mockOkHttpClient = getMockOkHttpClientMultipleRequest(getResourceFirstPageJson(), getResourceListJson());
+
+    final MockClient client = new MockClient("siteId", "apiKey", mockOkHttpClient);
+    Pager<MyResource> pager = client.listResources(null);
+    int count = 0;
+    for (MyResource res : pager) {
+      assertNotNull(res.getMyString());
+      count++;
+    }
+    assertEquals(4, count);
```

Comment:
I worry this test is not restrictive enough. Could we have different sizes and values in the pages so we can assert they are both being called and in the right order? Kind of like we do here: https://github.com/recurly/recurly-client-net/blob/3_0_0_beta/Recurly.Tests/PagerTest.cs#L41

I'm confident it will pass but we need to write these tests more like regression tests. Any small deviation in pagination behavior should fail the test.